dark_room_to_saliency.py

```diff
@@ -0,0 +1,59 @@
+import rospy
+from sensor_msgs.msg import Image
+from geometry_msgs.msg import Point
+from cv_bridge import CvBridge
+from attention import Saliency
+from std_msgs.msg import Float32MultiArray
+
+tensorflow_path = rospy.get_param("tensorflow_path", "/opt/tensorflow_venv/lib/python2.7/site-packages")
+model_file = rospy.get_param('~saliency_file', '/tmp/model.ckpt')
+network_input_height = float(rospy.get_param('~network_input_height', '192'))
+network_input_width = float(rospy.get_param('~network_input_width', '256'))
+
+@nrp.MapVariable("saliency", initial_value = Saliency(tensorflow_path, model_file, network_input_height, network_input_width, False))
+@nrp.MapVariable("saliency_pub", initial_value = rospy.Publisher("/saliency_map", Float32MultiArray, queue_size=1))
+@nrp.MapVariable("saliency_image_pub", initial_value = rospy.Publisher("/saliency_map_image", Image, queue_size=1))
+@nrp.MapVariable("bridge", initial_value=CvBridge())
+@nrp.MapVariable("last_time", initial_value = None)
+@nrp.MapVariable("elapsed", initial_value = 0)
+@nrp.MapVariable("published", initial_value = False)
+@nrp.MapRobotSubscriber("image", Topic("/hollie/camera/left/image_raw", Image))
+def image_to_saliency(t, saliency, saliency_pub, saliency_image_pub, bridge, image, last_time, elapsed, published):
+    if published.value:
+        return
+
+    if image.value is None:
+        return
+
+    if last_time.value is None:
+        last_time.value = t
+    current_time = t
+    dt = current_time - last_time.value
+    last_time.value = current_time
+
+    elapsed.value = elapsed.value + dt
+    if elapsed.value < 0.01:
+        return
+    else:
+        elapsed.value = 0.
+
+    image = bridge.value.imgmsg_to_cv2(image.value, "bgr8")
+
+    for i in range(0, len(image)):
+        for j in range(0, len(image[0])):
+            for k in range(0, len(image[0][0])):
+                image[i][j][k] = 0.
```

Comment:
Woa, you don't need to zero all the pixels, you can just create a new image of the same shape with:
np.zeros(saliency_map_shape)